### PR TITLE
Parity to noit, testing of check.ping

### PR DESCRIPTION
**Pull request by itzg**
For CMC-1248

### CODE REVIEW

**Comment by itzg** [comment]
I might need to mock the actual pinging part since Travis CI build produces this message, which I'd fully expect:

```
Error listening for ICMP packets: socket: permission denied
```

**Comment by rphillips** [comment]
ah hah. yeah, no root access on travis.

**Comment by itzg** [comment]
@rphillips, my first step was to detect if ping couldn't even execute, but now I'm left pondering if `PingCheck.Run` should return an error or indicate `CheckResultSet.SetStateUnavailable`. Thoughts?

**Comment by rphillips** [comment]
Return SetStateUnavailable would be really smart.

**Comment by itzg** [comment]
Ready for a re-review

File: check/check_ping.go [hunk 1]
```diff
@@ -82,14 +83,19 @@ func (ch *PingCheck) Run() (*CheckResultSet, error) {
...
-		metric.NewMetric("maximum", "", metric.MetricFloat, float64(stats.MaxRtt/time.Millisecond), "ms"),
-		metric.NewMetric("minimum", "", metric.MetricFloat, float64(stats.MinRtt/time.Millisecond), "ms"),
+		metric.NewMetric("maximum", "", metric.MetricFloat, utils.ScaleFractionalDuration(stats.MaxRtt, time.Second), metric.UnitSeconds),
+		metric.NewMetric("minimum", "", metric.MetricFloat, utils.ScaleFractionalDuration(stats.MinRtt, time.Second), metric.UnitSeconds),
 	)
 	crs := NewCheckResultSet(ch, cr)
-	crs.SetStateAvailable()
+	if stats.PacketsSent == 0 {
+		log.Warn("No ping packets were sent, likely due to lack of permission")
+		crs.SetStateUnavailable()
```
Comment (rphillips) [inline check/check_ping.go]: awesome!

File: check/check_test.go [hunk 2]
```diff
@@ -59,3 +61,68 @@ func TestCheckBase_GetWaitPeriod(t *testing.T) {
...
+		assert.Equal(t, m.Unit, actualM.Unit, "unit of %s", m.Name)
+
+	}
+}
+
+// Timebox is used for putting a time bounds around a chunk of code, given as the function boxed.
+// NOTE that if the duration d elapses, then boxed will be left to run off in its go-routine...it can't be
+// forcefully terminated.
+// This function can be used outside of a unit test context by passing nil for t
+func Timebox(t *testing.T, d time.Duration, boxed func(t *testing.T)) {
```
Comment (rphillips) [inline check/check_test.go]: awesome sauce

File: check/check_test.go [hunk 3]
```diff
@@ -59,3 +61,68 @@ func TestCheckBase_GetWaitPeriod(t *testing.T) {
...
+	timer := time.NewTimer(d)
+	completed := make(chan int)
```
Comment (rphillips) [inline check/check_test.go]: a nit... a recommended... guideline is to use a struct{} channel to pass a sentinel.

Comment (itzg) [inline check/check_test.go]: Thank you! I could not remember the proper way to do a sentinel channel. Then, I really just need to close the channel to trigger the select, right?

File: check/check_test.go [hunk 4]
```diff
@@ -108,21 +108,39 @@ func AssertMetrics(t *testing.T, expected []*ExpectedMetric, actual map[string]*
...
 
 	select {
 	case <-timer.C:
 		if t != nil {
 			t.Fatal("Timebox expired")
 		}
+		return false
 	case <-completed:
 		timer.Stop()
+		return true
```
Comment (rphillips) [inline check/check_test.go]: nice

**Comment by rphillips** [comment]
+1

**Comment by itzg** [comment]
@dimtruck, you're welcome to drop me more comments/suggestions, but I'm going to merge this down so @rphillips can make use of it.

**Pull request merged**